eessi/reframe/eessi-checks/applications/gromacs_check.py

```diff
@@ -54,27 +80,36 @@ def set_test_purpose(self):
     def skip_nb_impl_gpu_on_cpu_nodes(self):
         self.skip_if(
             (self.nb_impl == 'gpu' and not utils.is_gpu_present(self)),
-            "Skipping test variant with non-bonded interactions on GPUs, as this partition (%s) does not have GPU nodes" % self.current_partition.name
+            "Skipping test variant with non-bonded interactions on GPUs, "
+            f"as this partition ({self.current_partition.name}) does not have GPU nodes"
         )
 
     # Sckip testing when nb_impl=gpu and this is not a GPU build of GROMACS
     @run_after('setup')
     def skip_nb_impl_gpu_on_non_cuda_builds(self):
         self.skip_if(
             (self.nb_impl == 'gpu' and not utils.is_cuda_required(self)),
-            "Skipping test variant with non-bonded interaction on GPUs, as this GROMACS was not build with GPU support"
+            "Skipping test variant with non-bonded interactions on GPUs, "
+            f"as this module ({self.module_name}) was not build with GPU support"
         )
 
     # Skip testing GPU-based modules on CPU-based nodes
     @run_after('setup')
     def skip_gpu_test_on_cpu_nodes(self):
         hooks.skip_gpu_test_on_cpu_nodes(self)
 
-    # Assign num_tasks, num_tasks_per_node and num_cpus_per_task automatically based on current partition's num_cpus and gpus
+    # Assign num_tasks, num_tasks_per_node and num_cpus_per_task automatically
+    # based on current partition's num_cpus and gpus
     # Only when running nb_impl on GPU do we want one task per GPU
     @run_after('setup')
     def set_num_tasks(self):
-        if(self.nb_impl == 'gpu'):
-            hooks.assign_one_task_per_gpu(test = self, num_nodes = self.num_nodes)
+        if self.nb_impl == 'gpu':
+            hooks.assign_one_task_per_gpu(test=self, num_nodes=self.num_nodes)
         else:
-            hooks.assign_one_task_per_cpu(test = self, num_nodes = self.num_nodes)
+            hooks.assign_one_task_per_cpu(test=self, num_nodes=self.num_nodes)
+
+    @run_after('setup')
+    def set_omp_num_threads(self):
+        omp_num_threads = self.num_cpus_per_task
+        self.executable_opts += ['-dlb yes', f'-ntomp {omp_num_threads}', '-npme -1']
+        self.variables['OMP_NUM_THREADS'] = f'{omp_num_threads}'
```

Comment:
I know I didn't, but it's probably good to put a comment in the code as to why this line is here. It might seem redundant to set both the number of threads by passing `-ntomp` _and_ setting `OMP_NUM_THREADS`. The reason is that GROMACS produces an error if `-ntomp` and `OMP_NUM_THREADS` are set to conflicting values, which will result in a test failure (our system sets `OMP_NUM_THREADS` to some default value). The easiest way to avoid this is by explicitely making sure that both are set to the same value.

Reply:
Oh, also, ReFrame 4 gave me a deprecation warning on this. we should use `env_vars` instead of `variables` from now on.